app/src/main/java/com/dragic/gamehunter/view/home/HomeDeals.kt

```diff
@@ -12,19 +13,24 @@ import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
+import androidx.compose.runtime.snapshotFlow
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
 import androidx.compose.ui.res.dimensionResource
 import com.dragic.gamehunter.R
 import com.dragic.gamehunter.view.uicomponents.DealCard
 import com.dragic.gamehunter.view.uicomponents.ShimmerListItem
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.filter
 
 @Composable
 fun HomeDeals(
     deals: List<DealViewState>,
     onDealClick: (Int) -> Unit,
     modifier: Modifier = Modifier,
+    scrollState: LazyListState,
```

Comment:
Do you need to hoist `scrollState`? It looks like it can be contained in the Composable because you aren't doing anything with it in `HomeScreen` where it's constructed.